### PR TITLE
spellcheck: Add spelling linter

### PR DESCRIPTION
**Pull request by mariasfiraiala**
This is an attempt at adding a spell checker to the available actions. @gabrielmocanu You might find [the pattern for finding the md files](https://github.com/mariasfiraiala/actions/blob/9d7c7281d7eefe42fef5954127c6d8700d96e9bb/spellcheck/config/spellcheck.yml#L14) wrong. Apologies for that, I don't really have the means of testing this PR.

### CODE REVIEW

File: spellcheck/action.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+    - name: Spellcheck
+      uses: rojopolis/spellcheck-github-actions@v0
+      with:
+        config_path: config/spellcheck.yml
+        task_name: Markdown
```
Comment (gabrielmocanu) [inline spellcheck/action.yml:18]:
```suggestion
    - name: Checkout Code
      uses: actions/checkout@v3

    - name: Spellcheck
      uses: rojopolis/spellcheck-github-actions@0.33.1
      with:
        config_path: config/spellcheck.yml
        task_name: Markdown
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]
I don't think this path [1] will work, but we can test this after merging this PR.

[1]: https://github.com/security-summer-school/actions/pull/4/files#diff-6d93f5a4f7cde7d4e801ac15bc053890474f1aa9dc67cad6c9c7de45868989e6R17

**Comment by teodutu** [comment]
@mariasfiraiala you can test this before merging. Just create a repo, add this action, then raise a PR in that repo. If the action runs fine, paste a link to the run here and we'll know the config is good.

**Comment by mariasfiraiala** [comment]
@gabrielmocanu The path does not work, indeed. I've tested a couple of variations with it and I can't pin point exactly what's wrong. Any ideas?

**Comment by mariasfiraiala** [comment]
Manged to fix the config path problem. Tested it on [my repo](https://github.com/mariasfiraiala/essentials/actions/runs/5699873144/job/15449208485?pr=2), works like a charm. @gabrielmocanu Please take another look if possible

**Review by gabrielmocanu — APPROVED** [review]
Let's see how it's working.

**Pull request merged**